vllm/config.py

```diff
@@ -241,6 +241,12 @@ def verify_with_parallel_config(
                 "must be divisible by pipeline parallel size "
                 f"({pipeline_parallel_size}).")
 
+        if self.quantization == "bitsandbytes" and (
+                parallel_config.tensor_parallel_size > 1
+                or parallel_config.pipeline_parallel_size > 1):
+            raise ValueError(
+                "BitAndBytes quantization with TP or PP is not supported yet.")
```

Comment:
Is this easily done, or is there complexity within the quantization format that makes it difficult? It is worth noting for future work if so

Reply:
I understand that Pipeline Parallelism (PP) is currently disabled in vLLM, so I will not be focusing on it for now.

Tensor Parallelism (TP) should not be overly challenging, but it does require careful calibration of tensor sharding and concatenation, especially since the quantized tensor in bitsandbytes has a different shape than its original one. This is my first pull request for this community, and I aimed to keep it focused.

As I mentioned in the pull request description, parallelism will be my immediate next effort. I look forward to your understanding and support.

Reply:
Thanks for the clarity! Yes, TP would be nice as a followup in the near future